src/Sarif/Baseline/ResultMatching/DataStructures/ExtractedResult.cs

```diff
@@ -53,6 +81,19 @@ public bool MatchesAnyWhat(ExtractedResult other)
             return WhatComparer.MatchesWhat(this, other);
         }
 
+        /// <summary>
+        ///  Match enough of the 'What' properties of two ExtractedResults (Guid, Fingerprints, PartialFingerprints, Snippets, Message, Properties).
+        ///  A match in high-confidence identity properties is a match (Guid, Fingerprint, >= 50% of PartialFingerprint).
+        ///  A non-match in high-confidence identity properties is a non-match (Fingerprint, 0% of PartialFingerprints, Properties).
+        ///  Otherwise, Results match if Message and first Snippet match.
+        /// </summary>
+        /// <param name="other">ExtractedResult to match</param>
+        /// <returns>True if *any* 'What' property matches, False otherwise</returns>
+        internal bool MatchesAnyWhat(ExtractedResult other, TrustMap trustMap)
```

Comment:
>trustMap [](start = 69, length = 8)

How about defaulting `trustMap` to `null` and not having to add an overload?

Reply:
I had to have separate overloads because TrustMap, WhatMap, and WhatComponent are internal types, and I didn't really want to make them all public. If I make MatchesAnyWhat internal, however, it'll be a breaking change.

Reply:
Oh, I see -- you can't have a parameter of non-public type on a public API. I see now you've made the new overloads internal.

---
In reply to: [390454469](https://github.com/microsoft/sarif-sdk/pull/1807#discussion_r390454469) [](ancestors = 390454469)

Reply:
I don't think you need to be concerned about breaking changes. We can project non-trivial churn in this area, we've only just started to apply it ourselves, mostly to a single tool. 

---
In reply to: [390461605](https://github.com/microsoft/sarif-sdk/pull/1807#discussion_r390461605) [](ancestors = 390461605,390454469)